modules/programs/zimfw.nix

```diff
@@ -0,0 +1,157 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.zsh;
+
+  ## copied relToDotDir from ./zsh.nix. Probably should import it, but I don't
+  ## know how.
+  relToDotDir = file:
+    (optionalString (cfg.dotDir != null) (cfg.dotDir + "/")) + file;
+
+  ## zimfw doesn't exist in nixpkgs. I'm open to adding it there, but I'm not
+  ## quite sure how to configure it without home-manager, so the following
+  ## derivation may not be correct for a more generic case.
+  zimPkg = pkgs.stdenv.mkDerivation rec {
```

Comment:
Is there a PR to add it to nixpkgs? I think that we would want that done so that we don't maintain any package derivations in home-manager.